FirebaseAdmin/FirebaseAdmin.Tests/Messaging/InstanceIdClientTest.cs

```diff
@@ -60,5 +61,85 @@ public async Task UnsubscribeFromTopicAsync()
 
             Assert.Equal(1, result.SuccessCount);
         }
+
+        [Fact]
+        public async Task BadRequest()
+        {
+            var handler = new MockMessageHandler()
+            {
+                StatusCode = HttpStatusCode.BadRequest,
+                Response = "BadRequest",
+            };
+            var factory = new MockHttpClientFactory(handler);
+
+            var client = new InstanceIdClient(factory, MockCredential);
+
+            await Assert.ThrowsAsync<FirebaseMessagingException>(
+               () => client.SubscribeToTopicAsync("test-topic", new List<string> { "abc123" }));
+        }
+
+        [Fact]
+        public async Task Unauthorized()
+        {
+            var handler = new MockMessageHandler()
+            {
+                StatusCode = HttpStatusCode.Unauthorized,
+                Response = "Unauthorized",
+            };
+            var factory = new MockHttpClientFactory(handler);
+
+            var client = new InstanceIdClient(factory, MockCredential);
+
+            await Assert.ThrowsAsync<FirebaseMessagingException>(
+               () => client.SubscribeToTopicAsync("test-topic", new List<string> { "abc123" }));
```

Comment:
You should be able to run the following assertions on the exception:

```
Assert.Equal(ErrorCode.Unauthenticated, exception.ErrorCode);
Assert.Equal("Unexpected HTTP response with status: 401 (Unauthorized)\nUnauthorized", exception.Message);
Assert.Null(exception.MessagingErrorCode);
Assert.NotNull(exception.HttpResponse);
Assert.Null(exception.InnerException);
```

Similar assertions should be possible in the following test cases as well.